### ISSUE TITLE
Feature request: add variavble for GIR name to pkgconfig files

### ISSUE THREAD

**Issue by lw64** [mesonbuild/meson#11868]
The idea is to have every project roughly do this: https://gitlab.gnome.org/GNOME/libadwaita/-/merge_requests/770

According to @ebassi adding a feature to the pkgconfig generation to add the variable would be the best option. As input would be a value returned by gnome.generate_gir() used.

**Comment by xclaesse** [comment]
OOC, what's the use-case for having that `gir` variable? It is already common to have `girdir=${datadir}/gir-1.0`, `gobject-introspection-1.0.pc` itself has it. Meson already takes that variable from pc files to lookup for gir files. Wouldn't that be enough for you?

**Comment by eli-schwartz** [comment]
The difference here is that the pkg-config name is e.g. `libadwaita-1` while the gir name is `Adw-1`.

The goal seems to be to have a vala package depend on a pkg-config dependency and then use that dependency to map to an incompatible gir name.

This can of course be done by convention and an additional variable passed by manual construction to our pkg-config generator, but @lw64 is suggesting creating both those variables by passing some sort of `gir: mygir` kwarg to pkgconfig.generate() and have it read the install directory and gir name.

**Comment by xclaesse** [comment]
Adding that name mapping into the pkgconfig file won't be enough, you'll have to add it to your declare_dependency() as well for when it is a subproject, right?

**Comment by lw64** [comment]
Yes, that would make sense. Though currently also the vapi file is not directly accessible through a declared dependency. I think you need to do that manually by getting the right variable with the vapi. That would of course also work for the gir. I think it would be cool to make this also more convenient, but it would mostly require to add also a variable to declare_dependency for vapi metadata and maybe even custom code additions to export a complete binding.
